src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.GeneratedFileReplacingCompilationTracker.cs

```diff
@@ -31,7 +31,10 @@ private class GeneratedFileReplacingCompilationTracker(ICompilationTracker under
         private Compilation? _compilationWithReplacements;
 
         public ICompilationTracker UnderlyingTracker { get; } = underlyingTracker;
-        public SkeletonReferenceCache SkeletonReferenceCache { get; } = underlyingTracker.SkeletonReferenceCache.Clone();
+
+        public SkeletonReferenceCache? TryGetSkeletonReferenceCache() => UnderlyingTracker.TryGetSkeletonReferenceCache()?.Clone();
+        public SkeletonReferenceCache GetOrCreateSkeletonReferenceCache() => UnderlyingTracker.GetOrCreateSkeletonReferenceCache().Clone();
```

Comment:
⚠️ These are now going to clone the cache from the underlying tracker on _every call_ instead of just once

Reply:
Implicit fields backing property initializers are easy for me to miss for some reason. Will fix shortly.